java/memory/memory-core/src/main/java/org/apache/arrow/util/AutoCloseables.java

```diff
@@ -113,9 +114,23 @@ public static void close(Iterable<? extends AutoCloseable> ac) throws Exception
    */
   @SafeVarargs
   public static void close(Iterable<? extends AutoCloseable>...closeables) throws Exception {
-    close(Arrays.asList(closeables).stream()
-        .flatMap((Iterable<? extends AutoCloseable> t) -> Collections2.toList(t).stream())
-        .collect(Collectors.toList()));
+    close(flatten(closeables));
+  }
+
+  @SafeVarargs
+  private static Iterable<AutoCloseable> flatten(Iterable<? extends AutoCloseable>... closeables) {
+    return new Iterable<AutoCloseable>() {
+      // Cast from Iterable<? extends AutoCloseable> to Iterable<AutoCloseable> is safe in this context
+      // since there's no modification of the original collection
+      @SuppressWarnings("unchecked")
+      @Override
+      public Iterator<AutoCloseable> iterator() {
+        return Arrays.stream(closeables)
+            .flatMap((Iterable<? extends AutoCloseable> i)
+                -> StreamSupport.stream(((Iterable<AutoCloseable>) i).spliterator(), false))
```

Comment:
could you provide a parameter comment for the false literal?

Reply:
Hopefully addressed by latest commit

Reply:
```suggestion
                -> StreamSupport.stream(((Iterable<AutoCloseable>) i).spliterator(), /*parallel=*/false))
```